plugin/pkg/auth/authenticator/password/keystone/keystone.go

```diff
@@ -39,11 +39,30 @@ type KeystoneAuthenticator struct {
 
 // AuthenticatePassword checks the username, password via keystone call
 func (keystoneAuthenticator *KeystoneAuthenticator) AuthenticatePassword(username string, password string) (user.Info, bool, error) {
-	opts := gophercloud.AuthOptions{
+	var opts gophercloud.AuthOptions
+	opts = gophercloud.AuthOptions{
 		IdentityEndpoint: keystoneAuthenticator.authURL,
 		Username:         username,
 		Password:         password,
 	}
+	if strings.HasSuffix(keystoneAuthenticator.authURL, "/v3") {
+		// username has a format of domain\name.
+		firstIndex := strings.Index(username, "\\")
```

Comment:
- The format of "domain\name" is not a new custom domain encoding scheme but widely used by AD based systems where a keystone domain is a AD domain actually. The AD domains usually follow NetBIOS domain name format(https://support.microsoft.com/en-us/kb/909264) which does not allow backslash.
- username/password is useful when  kubectl users do not understand oepnstack and just want to consume k8s(which is hosted on openstack).